File: task-service/src/main/java/com/manabie/todo/entity/TaskEntity.java
package com.manabie.todo.entity;

import com.manabie.todo.constant.TaskStatus;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "task")
@Getter
@Setter
public class TaskEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "task_id_seq")
    @SequenceGenerator(name = "task_id_seq", sequenceName = "task_id_seq", allocationSize = 1)
    private Long id;
    private String title;
    private String description;
    private Long owner;
    private TaskStatus status;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;
}

File: domain/src/main/kotlin/com/example/domain/model/TaskEntity.kt
package com.example.domain.model

import java.time.LocalDateTime
import java.util.*

data class TaskEntity(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val description: String,
    val userId: String,
    var createdAt: LocalDateTime = LocalDateTime.now(),
    var updatedAt: LocalDateTime = LocalDateTime.now(),
)
